xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletRequest.java

```diff
@@ -101,4 +107,10 @@ public String getRemoteHost()
 
         return request.getRemoteHost();
     }
+
+    @Override
+    public Optional<UserReference> getEffectiveAuthor()
+    {
+        return Optional.ofNullable((UserReference) getAttribute(ATTRIBUTE_EFFECTIVE_AUTHOR));
```

Comment:
isn't this cast possibly dangerous? There's no way the value associated to that key is replaced by a something else by mistake?

Reply:
Also, it's hard to imagine someone using something as complex as the value of ATTRIBUTE_EFFECTIVE_AUTHOR for something completely different.

What protection would you have in mind, return Optional.empty() if the type is not good ?

Honestly, getting an exception when a bad object was put there does not sound so bad to me, at least you know why it does not work. We have this kind of assumption pretty much everywhere we manipulate this kind of object map (you are going to find this kind of cast before pretty much every call to ExecutionContext#getProperty for example).

Reply:
> What protection would you have in mind, return Optional.empty() if the type is not good ?

Yes, that's what I was thinking about.

Reply:
> Yes, that's what I was thinking about.

But that would hide the error. Or do you also suggest logging a warning. Same as @tmortagne, I think throwing an error is better in this case because we're not expecting the value to have a different type.